src/screens/settings/components/SettingsContainer.js

```diff
@@ -0,0 +1,224 @@
+
+// Utils
+import {
+  getUser,
+  setUser,
+  clearWordList,
+  getUserWordList,
+  getAllUsers,
+} from '../../../utils/storage';
+
+// Context
+import UserContext from '../../../contexts/UserContext';
+
+// Styles
+const styles = StyleSheet.create({
+  wordListContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    padding: 5,
+  },
+  wordList: {
+    flex: 5,
+    flexDirection: 'row',
+  },
+  refresh: {
+    flex: 1,
+    flexDirection: 'row',
+  },
+  wordLevel: {
+    padding: 15,
+  },
+});
+
+// Vars
+const NEW_USER = 'New User';
+
+const updateData = (inputText, setName) => {
+  setName(inputText);
+  setUser(inputText);
+};
+
+const clearWords = async () => {
+  clearWordList();
+};
+
+const handleClearButtonPress = () => {
+  Alert.alert(
+    'Are you sure you want to delete all of your words?',
+    'This action is irreversible. You will lose all of your word data.',
+    [
+      {
+        text: 'Cancel',
+        onPress: () => {},
+        style: 'cancel',
+      },
+      { text: 'Delete All Words', onPress: clearWords, style: 'danger' },
+    ],
+  );
+};
+
+const getUsers = async () => {
+  const users = await getAllUsers();
+  return users;
+};
+
+const SettingsContainer = () => {
+  const [name, setName] = useContext(UserContext);
+  const [inputText, setInputText] = useState('');
+  const [userData, setUserData] = useState({});
+  const [users, setUsers] = useState([]);
+  const [inputVisible, setInputVisible] = useState(false);
+
+  const refreshList = useCallback(() => {
+    getUserWordList(name).then(setUserData);
+  }, [name]);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, [setUsers]);
+
+  useEffect(() => {
+    refreshList();
+  }, [refreshList]);
+
+  useEffect(() => {
+    setInputVisible(false);
+    setUser(name);
+  }, [name]);
+
+  const logOut = () => {
+    setUser(null);
+    setName(null);
+  };
+
+  const getWords = () => {
+    const levels = Object.keys(userData || {});
+    if (levels.length) {
+      const words = levels.reduce(
+        (obj, level) => ({
+          ...obj,
+          [level]: Object.keys(userData[level]),
+        }),
+        {},
+      );
+
+      return words;
+    }
+
+    return {};
+  };
+
+  const changeUser = (value) => {
+    if (value !== NEW_USER) {
+      setName(value);
+    } else {
+      setInputVisible(true);
+    }
+  };
+
+  const getPickerItems = () => {
+    const pickers = [...users, NEW_USER].map((user) => (
+      <Picker.Item label={user} value={user} />
+    ));
+    return pickers;
+  };
+
+  const getWordLevels = () => {
+    const words = getWords();
+
+    const listWords = (str, word) => `${str}, ${word}`;
+
+    const components = Object.keys(words).map((level) => (
+      <View style={styles.wordLevel}>
+        <H3>
+          {level[0].toUpperCase()}
+          {level.slice(1)}
+        </H3>
+        <Text>{words[level].reduce(listWords, '').slice(2)}</Text>
+      </View>
+    ));
+
+    return components;
+  };
```

Comment:
There's a lot going on here. Would it maybe make sense to break this component apart into a few different pieces? Is there a logical way to do so?

Re: effects - there are 3 effects here and it isn't super clear what each is doing.

The first one appears to be fetching some user data whenever this component mounts.
The second one is calling `getUserWordList` whenever `name` changes if I understand correctly.
The last one hides the input and calls `setUser(name)` whenever `name` changes.

Even given that I'm having a hard time following what that _means_. That might be ok, as long as it's clear to you (or anyone more familiar with the domain/app) that's fine.

Reply:
Got it. I'll add this to my todo list and add it to another PR